test/index.js

```diff
@@ -253,6 +255,53 @@ describe('onBuild()', () => {
     expect(config.config.env.NEXTAUTH_URL).toEqual(mockUserDefinedSiteUrl)
   })
 
+  test('sets the NEXTAUTH_URL to the DEPLOY_PRIME_URL when CONTEXT env variable is not \'production\'', async () => {
+    const mockUserDefinedSiteUrl = chance.url()
+    process.env.DEPLOY_PRIME_URL = mockUserDefinedSiteUrl
+    process.env.URL = chance.url()
+    
+    // See https://docs.netlify.com/configure-builds/environment-variables/#build-metadata for all possible values
+    process.env.CONTEXT = 'deploy-preview'
+
+    await moveNextDist()
+
+    const initialConfig = await getRequiredServerFiles(netlifyConfig.build.publish)
+
+    initialConfig.config.env.NEXTAUTH_URL = mockUserDefinedSiteUrl
+    await updateRequiredServerFiles(netlifyConfig.build.publish, initialConfig)
+
+    await nextRuntime.onBuild(defaultArgs)
+
+    expect(onBuildHasRun(netlifyConfig)).toBe(true)
+    const config = await getRequiredServerFiles(netlifyConfig.build.publish)
+
+    expect(config.config.env.NEXTAUTH_URL).toEqual(mockUserDefinedSiteUrl)
+  })
+  
+  test('sets the NEXTAUTH_URL to the URL when CONTEXT env variable is \'production\'', async () => {
```

Comment:
[dust] Be more specific which URL.

```suggestion
  test('sets the NEXTAUTH_URL to the user defined site URL when CONTEXT env variable is \'production\'', async () => {
```